net: lib: mqtt_helper: Add support for IPv6 broker address (#10519)
nrfconnect/sdk-nrf: samples/net/mqtt/doc/description.rst

     west build -b nrf9160dk_nrf9160_ns -- -DOVERLAY_CONFIG=overlay-tls-nrf9160.conf

> Or for the Thingy:91 with TLS and `mqtt_helper` debug enabled (shown in the output below):

Reviewer: ```suggestion
Or for the Thingy:91 with TLS and ``mqtt_helper`` debug enabled (shown in the following :ref:`sample output <mqtt_sample_output_IPv6>`):
```

Author: Changed

      [00:04:01.475,982] <inf> transport: Publishing message: "Hello MQTT! Current uptime is: 241458" on topic: "my/publish/topic"
      [00:05:01.475,982] <inf> transport: Publishing message: "Hello MQTT! Current uptime is: 301459" on topic: "my/publish/topic"

> Sample output using LTE with IPv6:

Reviewer: ```suggestion
The following sample output is displayed using LTE with IPv6:
```
Do we need to follow the same testing steps mentioned in Testing section to get this sample output (using LTE with IPv6)?

Author: Yes, same testing steps, just with a different build configuration/device. I added the details of the specific build command (i.e. for the Thingy:91) that matches the sample output.

Different configurations will have slightly different output, but they all now support IPv6, i.e. IPv6 will now also work with Wi-Fi. Actually, now that I have made the IP address logging at DBG level, the output will not look any different than it did before, unless you turn on that additional level.

> Sample output using LTE with IPv6:

Author: Additional details added to the Wi-Fi example description (which also had TLS enabled).

.. code-block:: console

      *** Booting Zephyr OS build v3.2.99-ncs2 ***
      [00:00:00.500,061] <inf> network: Connecting to LTE...
      [00:00:03.022,644] <inf> network: PDN connection activated

>       [00:00:08.429,473] <inf> mqtt_helper: IPv6 Address found 2001:41d0:1:925e::1 (AF_INET6)

Author: I like this idea, as if the example is supposed to be showing MQTT, then increasing the log level for MQTT, just for that example, will help the reader see what is happening.

>       [00:00:08.429,473] <inf> mqtt_helper: IPv6 Address found 2001:41d0:1:925e::1 (AF_INET6)

Author: Example updated with the build command and mqtt_help debug output

Sample output using LTE with IPv6:

.. code-block:: console

      *** Booting Zephyr OS build v3.2.99-ncs2 ***
      [00:00:00.500,061] <inf> network: Connecting to LTE...
      [00:00:03.022,644] <inf> network: PDN connection activated
      [00:00:08.338,256] <inf> mqtt_helper: IPv6 Address found 2001:41d0:1:925e::1
      [00:00:11.950,714] <inf> transport: Connected to MQTT broker
      [00:00:11.950,775] <inf> transport: Hostname: test.mosquitto.org
      [00:00:11.950,805] <inf> transport: Client ID: 350457791735879
      [00:00:11.950,805] <inf> transport: Port: 8883
      [00:00:11.950,836] <inf> transport: TLS: Yes
      [00:00:12.784,667] <inf> transport: Subscribed to topic my/subscribe/topic
      [00:00:12.784,973] <inf> transport: Received payload: $ on topic: my/subscribe/topic
      [00:01:00.492,706] <inf> transport: Published message: "Hello MQTT! Current uptime is: 60491" on topic: "my/publish/topic"
      [00:01:10.332,122] <inf> transport: Received payload: Test message on topic: my/subscribe/topic
      [00:02:00.501,220] <inf> transport: Published message: "Hello MQTT! Current uptime is: 120491" on topic: "my/publish/topic"

> 

Reviewer 2: Extra empty line.

Author: Removed

      [00:04:01.475,982] <inf> transport: Publishing message: "Hello MQTT! Current uptime is: 241458" on topic: "my/publish/topic"
      [00:05:01.475,982] <inf> transport: Publishing message: "Hello MQTT! Current uptime is: 301459" on topic: "my/publish/topic"

> Sample output showing IPv6, following the same instructions but for a different build configuration, using LTE on the Thingy:91, with the TLS overlay, and debug logging enabled for the MQTT helper library enabled:

Reviewer: ```suggestion
.. _mqtt_sample_output_IPv6:

The sample output showing IPv6, following the same instructions but for a different build configuration, using LTE on the Thingy:91, with the TLS overlay, and debug logging enabled for the :ref:`lib_mqtt_helper` library:
```

Author: Changed
